src/Microsoft.DotNet.Wpf/src/PresentationBuildTasks/Microsoft/Build/Tasks/Windows/CreateTemporaryTargetAssemblyProject.cs

```diff
@@ -0,0 +1,414 @@
+
+        }
+
+        //
+        // Add a list of files into an Item in the project file, the ItemName is specified by sItemName.
+        //
+        private void AddNewItems(XmlDocument xmlProjectDoc, string sItemName, ITaskItem[] pItemList)
+        {
+            if (xmlProjectDoc == null || String.IsNullOrEmpty(sItemName) || pItemList == null)
+            {
+                // When the parameters are not valid, simply return it, instead of throwing exceptions.
+                return;
+            }
+
+            XmlNode root = xmlProjectDoc.DocumentElement;
+
+            // Create a new ItemGroup element
+            XmlNode nodeItemGroup = xmlProjectDoc.CreateElement(ITEMGROUP_NAME, root.NamespaceURI);
+
+            // Append this new ItemGroup item into the list of children of the document root.
+            root.AppendChild(nodeItemGroup);
+
+            XmlElement embedItem = null;
+
+            for (int i = 0; i < pItemList.Length; i++)
+            {
+                // Create an element for the given sItemName
+                XmlElement nodeItem = xmlProjectDoc.CreateElement(sItemName, root.NamespaceURI);
+
+                // Create an Attribute "Include"
+                XmlAttribute attrInclude = xmlProjectDoc.CreateAttribute(INCLUDE_ATTR_NAME);
+
+                ITaskItem pItem = pItemList[i];
+
+                // Set the value for Include attribute.
+                attrInclude.Value = pItem.ItemSpec;
+
+                // Add the attribute to current item node.
+                nodeItem.SetAttributeNode(attrInclude);
+
+                if (TRUE == pItem.GetMetadata(EMBEDINTEROPTYPES))
+                {
+                    embedItem = xmlProjectDoc.CreateElement(EMBEDINTEROPTYPES, root.NamespaceURI);
+                    embedItem.InnerText = TRUE;
+                    nodeItem.AppendChild(embedItem);
+                }
+
+                string aliases = pItem.GetMetadata(ALIASES);
+                if (!String.IsNullOrEmpty(aliases))
+                {
+                    embedItem = xmlProjectDoc.CreateElement(ALIASES, root.NamespaceURI);
+                    embedItem.InnerText = aliases;
+                    nodeItem.AppendChild(embedItem);
+                }
+
+                // Add current item node into the children list of ItemGroup
+                nodeItemGroup.AppendChild(nodeItem);
+            }
+        }
+
+        private void AddNewProperties(XmlDocument xmlProjectDoc, List<(string PropertyName, string PropertyValue)> properties )
+        {
+            if (xmlProjectDoc == null || properties == null )
+            {
+                // When the parameters are not valid, simply return it, instead of throwing exceptions.
+                return;
+            }
+
+            XmlNode root = xmlProjectDoc.DocumentElement;
+
+            // Create a new ItemGroup element
+            XmlNode nodeItemGroup = xmlProjectDoc.CreateElement("PropertyGroup", root.NamespaceURI);
```

Comment:
Yes, this is required because the PackageReference NuGet props/targets can hook in to the temp project build and build other projects, etc. after the temp project has defined immutable global proeprties.  These globals interfere with other properties of the same name and cause unexpected behavior with the build.  One of them was 'AssemblyName': very likely to collide with other projects.